machines/VCItemMachine/ExistingMosipVCItem/ExistingMosipVCItemMachine.ts

```diff
@@ -787,11 +787,12 @@ export const ExistingMosipVCItemMachine =
         }),
 
         setStoreVerifiableCredential: model.assign((context, event) => {
+          const eventResponse = event?.response ? event.response : event?.vc;
           return {
             ...context,
-            ...event.vc,
+            ...eventResponse,
             storeVerifiableCredential: {
-              ...event.vc.verifiableCredential,
+              ...eventResponse.verifiableCredential,
```

Comment:
Please add the comment with Bala's findings here so that it helps us in future when we re-look at this code.